File: pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{

    int coinCount = 0;
    float change;

    do
    {
        change = get_float("Change owed: ");
    }
    while (change <= 0);

    int cents = round(change * 100);

    do
    {
        coinCount++;

        if (cents >= 25)
        {
            cents = cents - 25;
        }
        else if (cents >= 10)
        {
            cents = cents - 10;
        }
        else if (cents >= 5)
        {
            cents = cents - 5;
        }
        else
        {
            cents = cents - 1;
        }
    }
    while (cents > 0);

    printf("%i\n", coinCount);

}

File: pset6/cash.py
from cs50 import get_float

def main():
    change = get_valid_float()
    print(calculate_coint_count(change))

def calculate_coint_count(change):
    coinCount = 0
    cents = change * 100
    while True:
        if cents >= 25:
            cents -= 25
        elif cents >= 10:
            cents -= 10
        elif cents >= 5:
            cents -= 5
        elif cents >= 1:
            cents -= 1
        else:
            break
        coinCount += 1
    return coinCount

def get_valid_float():
    while True:
        n = get_float("Change owned: ")
        if n > 0:
            break
    return n

main()